### PR TITLE
Virtual Defund Skeleton

### PR DESCRIPTION
**Pull request by lalexgap**
Creates some basic infrastructure for the virtual defunding Objective.

The PR implements a basic virtual defunding objective that satisfies the `Objective` interface.

### CODE REVIEW

**Review by andrewgordstewart — APPROVED** [review]

File: protocols/virtualdefund/virtualdefund.go [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+
+// ObjectiveRequest represents a request to create a new virtual defunding objective.
+type ObjectiveRequest struct {
+	VirtualChannelId types.Destination
+}
+
+// Objective contains relevent information for the defund objective
+type Objective struct {
+	Status protocols.ObjectiveStatus
+	V      *channel.SingleHopVirtualChannel
```
[the remark this answers is not in the document]
Comment (lalexgap) [reply protocols/virtualdefund/virtualdefund.go:27]: Let me give it a go! Perhaps I can merge this PR in and try it it a new PR?

File: protocols/virtualdefund/virtualdefund.go [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+	relatable := []protocols.Storable{o.V}
+
+	if o.ToMyLeft != nil {
+		relatable = append(relatable, o.ToMyLeft)
+	}
+
+	if o.ToMyRight != nil {
+		relatable = append(relatable, o.ToMyRight)
+	}
+	return relatable
```
[the remark this answers is not in the document]
Comment (lalexgap) [reply protocols/virtualdefund/virtualdefund.go:70]: `relatable` is initialized with `V` `[]protocols.Storable{**o.V**}`

File: protocols/virtualdefund/virtualdefund.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (andrewgordstewart) [reply protocols/virtualdefund/virtualdefund.go:27]: that sounds reasonable!

File: protocols/virtualdefund/virtualdefund.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (andrewgordstewart) [reply protocols/virtualdefund/virtualdefund.go:70]: ah, yes!

**Pull request merged**